src/term.rs

```diff
@@ -10,11 +10,24 @@ use std::os::windows::io::{AsRawHandle, RawHandle};
 
 use crate::{kb::Key, utils::Style};
 
+trait TermWrite: Write + fmt::Debug + AsRawFd + Send {}
+impl<T: Write + fmt::Debug + AsRawFd + Send> TermWrite for T {}
+
+trait TermRead: Read + fmt::Debug + AsRawFd + Send {}
+impl<T: Read + fmt::Debug + AsRawFd + Send> TermRead for T {}
+
+#[derive(Debug, Clone)]
+pub struct ReadWritePair {
+    read: Arc<Mutex<dyn TermRead>>,
+    write: Arc<Mutex<dyn TermWrite>>,
+}
+
 /// Where the term is writing.
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
```

Comment:
The derivation of `Copy` just isn't possible in a safe way for arbitrary `Read`/`Write` destinations. And allowing for equality checks is of questionable value, although we can add it back if need be.

Reply:
I would add it back to maintain as much backward compatibility as possible.

Reply:
`Copy` cannot be added back, since no useful file handles are `Copy` (without `unsafe`).

`Eq`/`PartialEq` might be possible to add back, but it would need a strange/custom implementation (ie, assigning the `Term` a unique id?), because neither of those traits is object safe:
```
error[E0038]: the trait `TermWrite` cannot be made into an object
  --> src/term.rs:28:22
   |
28 |     write: Arc<Mutex<dyn TermWrite>>,
   |                      ^^^^^^^^^^^^^ `TermWrite` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
  --> src/term.rs:15:44
   |
15 | trait TermWrite: Write + Debug + AsRawFd + PartialEq + Send {}
   |       ---------                            ^^^^^^^^^ ...because it uses `Self` as a type parameter
   |       |
   |       this trait cannot be made into an object...
```